07_errors/hsy3418/store_test.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+func (suite *storerTestSuite) SetupTest() {
+	suite.store = suite.makeStore()
+	suite.toBeCreatedPuppy = Puppy{ID: 101, Breed: "Poodle", Colour: "White", Value: 1000.5}
+	suite.existsPuppy = Puppy{ID: 102, Breed: "Poodle", Colour: "White", Value: 1280.5}
+	suite.toBeUpdatedPuppy = Puppy{ID: 102, Breed: "Poodle", Colour: "White", Value: 2000}
+	suite.invalidPuppy = Puppy{ID: 103, Breed: "Poodle", Colour: "White", Value: -1000}
+	suite.invaildError = ErrorEf(ErrInvalidInput, "The puppy value is invalidate")
+	suite.invalidIDError = ErrorEf(ErrNotFound, "This puppy does not exist")
+	suite.dupicatesError = ErrorEf(ErrDuplicate, "This puppy exists ")
+	err := suite.store.CreatePuppy(suite.existsPuppy)
+	if err != nil {
+		suite.FailNow("Failed to setup test")
+	}
+}
+
+type storerTestSuite struct {
+	suite.Suite
+	store            Storer
+	makeStore        func() Storer
+	toBeCreatedPuppy Puppy
+	existsPuppy      Puppy
+	toBeUpdatedPuppy Puppy
+	invalidPuppy     Puppy
+	invaildError     error
```

Comment:
These three errors are constant and only used for comparative assertions. You should make these constants, having global scope and then you won't have to have the assignment code within the `SetupTest()` method.

```go
const (
  ErrorInvalidInput = ErrorEf(ErrInvalidInput, "The puppy value is invalidate")
  ErrorNotFound = ErrorEf(ErrNotFound, "This puppy does not exist")
  ErrorDuplicate = ErrorEf(ErrDuplicate, "This puppy exists ")
)
```